### PR TITLE
Admin profile

### PR DESCRIPTION
**Pull request by fahadansari12**

### CODE REVIEW

File: app/Http/Controllers/AdminEditController.php [hunk 1]
```diff
@@ -16,15 +16,15 @@ public function index() {
 
     
         public function updateProfile(Request $req) {
-            $name =  Session::get('admin-name');
-            Session::flush();
-            
+            $p = Auth::user()->password;
+            return dd($p);
```
Comment (Rahul-Khera-Codes) [inline app/Http/Controllers/AdminEditController.php]: what is $p and dd??

File: app/Http/Controllers/UserProfileController.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        ]); 
+
+        $date = $request->input('date-of-report');
+
+        $technology = $request->input('technology');
+        $mentor = $request->input('mentor');
+        $report = $request->input('report-details');
+        $status = $request->input('status-of-report');
+
+        DB::table('user-report')->insert(
```
Comment (Rahul-Khera-Codes) [inline app/Http/Controllers/UserProfileController.php]: create a function in model called insertUser, don't do direct db here

File: app/User.php [hunk 3]
```diff
@@ -18,6 +18,11 @@ class User extends Authenticatable
         'name', 'email', 'password',
     ];
 
+    public function getMobileAttribute($value)
```
Comment (Rahul-Khera-Codes) [inline app/User.php]: what is use of this?

File: app/Providers/RouteServiceProvider.php [hunk 4]
```diff
@@ -39,6 +39,10 @@ public function map()
 
         $this->mapWebRoutes();
 
+        $this->mapUserRoutes();
```
Comment (Rahul-Khera-Codes) [inline app/Providers/RouteServiceProvider.php]: what is use of Providers? can u explain. i am not aware

File: app/Http/Middleware/IsAdmin.php [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    /**
+     * Handle an incoming request.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \Closure  $next
+     * @return mixed
+     */
+    public function handle($request, Closure $next)
+    {
+        if (Auth::user() &&  Auth::user()->admin == 1) {
```
Comment (Rahul-Khera-Codes) [inline app/Http/Middleware/IsAdmin.php]: why == 1  

why is this not boolean?

File: app/Http/Controllers/UserProfileController.php (hunk 2 above)
Comment (fahadansari12) [reply app/Http/Controllers/UserProfileController.php]: ok sir, I am working on it.

File: app/User.php (hunk 3 above)
Comment (fahadansari12) [reply app/User.php]: I was trying to use it as an alternate to session, but it didn't worked.

File: app/Providers/RouteServiceProvider.php (hunk 4 above)
Comment (fahadansari12) [reply app/Providers/RouteServiceProvider.php]: sir, providers are generally used to configure the application and register various functionality including routing. so, here I am using it to register new route file.

File: app/Http/Middleware/IsAdmin.php (hunk 5 above)
Comment (fahadansari12) [reply app/Http/Middleware/IsAdmin.php]: it is used as boolean value, sir.

File: app/Http/Controllers/AdminEditController.php (hunk 1 above)
Comment (fahadansari12) [reply app/Http/Controllers/AdminEditController.php]: sir, dd() is a dump and die function. which is used to halt further execution and return value inside it. It is generally used to check for the errors. I mistakenly forgot to remove that.

**Pull request merged**